File: datasets/php/HumanEval_45_triangle_area.php
<?php
// Given length of a side and high return area for a triangle.
// >>> triangle_area(5, 3)
// 7.5
function triangle_area($a, $h) {
	return ($a * $h) / 2;
}


function candidate(...$args) {
    return triangle_area(...$args);
}

function test(): void {
    if (candidate(5, 3) !== 7.5) { throw new Exception("Test failed!"); }
    if (candidate(2, 2) !== 2.0) { throw new Exception("Test failed!"); }
    if (candidate(10, 8) !== 40.0) { throw new Exception("Test failed!"); }
}

test();

File: datasets/rb/HumanEval_45_triangle_area.rb
# Given length of a side and height return area for a triangle.
# >>> triangle_area(5, 3)
# 7.5
def triangle_area(a, h)
  a * h / 2
end

require 'test/unit'
class TestHumanEval < Test::Unit::TestCase
  def test_triangle_area
    candidate = method(:triangle_area)
    assert_equal(7.5, candidate.call(5, 3))
    assert_equal(2.0, candidate.call(2, 2))
    assert_equal(40.0, candidate.call(10, 8))
  end
end
